rasa/nlu/extractors/lookup_entity_extractor.py

```diff
@@ -0,0 +1,119 @@
+        #      person: /some/other/path/person.txt
+        "lookup": None
+    }
+
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None):
+        super(LookupEntityExtractor, self).__init__(component_config)
+
+        if component_config is not None and "lookup" in component_config:
+            if component_config["lookup"] is not None:
+                # check if the entities and respective file path exists
+                for key, value in list(component_config["lookup"].items()):
+                    self.add_entities_if_lookup_entity(key, value)
+            else:
+                self.component_config["lookup"] = None
+
+            if not bool(component_config["lookup"]):
+                # if the lookup dictionary is empty after filtering
+                # the lookup entities, assign it None value
+                self.component_config["lookup"] = None
+        else:
+            common_utils.raise_warning(
+                "Can't extract Lookup Entities,\
+                Please configure the lookup entities in the config.yml."
+            )
+
+    def add_entities_if_lookup_entity(self, entity, file_path) -> None:
+        if file_path is not None:
+            if os.path.isfile(file_path):
+                pass
+            else:
+                # remove the entity from the lookup dictionary,
+                # if the file path doesn't exist
+                self.component_config["lookup"].pop(entity)
+                common_utils.raise_warning(
+                    f"can't extract lookup entity: '{entity}',\
+                    make sure the provided file '{file_path}' exists.")
+        else:
+            # remove the entity from the lookup dictionary,
+            # if the file path is not provided
+            self.component_config["lookup"].pop(entity)
+            common_utils.raise_warning(
+                f"can't extract '{entity}' entity,\
+                please provide the example file.")
+
+    def _parse_entities(self, user_input: Text) -> List[Dict[Text, Any]]:
+        """Extract entities from the user input."""
+        if self.component_config["lookup"] is not None:
+            for entity, file_path in list(
+                    self.component_config["lookup"].items()):
+                results = self._parse_all_entities(
+                    user_input, entity, file_path)
+                return results
+        else:
+            return []
+
+    @staticmethod
+    def _parse_all_entities(
+        user_input: str, entity: list, file_path: list
+    ) -> List[Dict[Text, Any]]:
+        """
+        This method does the actual entity extraction work.
+        So here I am running the loop over the list of data in the text file
+        and check whether it exists in the user's message
+        """
+        results = []
+        with open(file_path, "r") as f:
+            examples = f.readlines()
+
+            for example in examples:
+                example = example.lower().strip()
```

Comment:
Let's add an option `lowercase`. We can set its default value to `True`. But I think it might happen that users need to consider the casing when using this component.